src/mscorlib/src/System/Diagnostics/Eventing/XplatEventLogger.cs

```diff
@@ -79,6 +79,10 @@ private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnly
                 return String.Empty;
 
             Contract.Assert(payloadName.Count == payload.Count);
+            if(payloadName.Count != payload.Count)
```

Comment:
So if the asset is still valid then the process will still crash but on debug builds only. Do you know what is the actual reason that causes the caller of this function to pass incorrect data?